AlertaDengue/upload/tasks.py

```diff
@@ -1,438 +1,328 @@
-
-        for chunk, (lowerbound, upperbound) in enumerate(
-            chunk_gen(10000, dbf.numrec)
-        ):
-            parquet_fname = os.path.join(
-                str(sinan.chunks_dir), f"{dbf_name}-{chunk}.parquet"
-            )
-            df = gpd.read_file(
-                str(sinan.filepath),
-                # include_fields=list(EXPECTED_FIELDS.values()),
-                rows=slice(lowerbound, upperbound),
-                ignore_geometry=True,
-            )
-
-            missing_cols = set(EXPECTED_FIELDS.values()).difference(
-                set(df.columns)
+            if not col in columns:
+                if synonym in columns:
+                    for i, req in enumerate(sinan.REQUIRED_COLS):
+                        if req == col:
+                            sinan.REQUIRED_COLS[i] = synonym
+
+                    cols = sinan.COLUMNS.copy()
+                    for column, final_col in sinan.COLUMNS:
+                        if column == col:
+                            del cols[column]
+                            cols[synonym] = final_col
+                    sinan.COLUMNS = cols
+                    sinan.save()
+        except:
+            sinan.status.warning(
+                f"Could not use the synonym '{synonym}' of the field '{col}'"
             )
 
-            for col in missing_cols:
-                df[col] = None
+    if not all(col in columns for col in sinan.REQUIRED_COLS):
+        missing_cols = set(sinan.REQUIRED_COLS).difference(set(columns))
+        err = f"Missing required columns: {missing_cols}"
+        raise SINANUploadFatalError(sinan.status, err)
 
-            df.to_parquet(parquet_fname)
+    if not all(col in columns for col in sinan.COLUMNS):
+        missing_cols = set(sinan.COLUMNS).difference(set(columns))
+        warning = f"Missing columns (filled with <NA>): {missing_cols}"
+        sinan.status.warning(warning)
 
-        sinan.status = Status.WAITING_INSERT
-        sinan.save(update_fields=["status"])
-        return True
+    sinan.status.debug("Task 'sinan_verify_file' finished.")
 
 
-@shared_task(default_retry_delay=30, max_retries=3, soft_time_limit=3 * 60)
-def chunk_parquet_file(sinan_pk: int) -> bool:
-    sinan = SINAN.objects.get(pk=sinan_pk)
+def insert_chunk_to_temp_table(
+    upload_sinan_id: int,
+    df_chunk: pd.DataFrame,
+    tablename: str,
+    cursor,
+):
+    sinan = SINANUpload.objects.get(pk=upload_sinan_id)
+    columns = list(sinan.COLUMNS.values())
 
-    if sinan.status == Status.WAITING_CHUNK:
-        sinan.status = Status.CHUNKING
-        sinan.save(update_fields=["status"])
+    print(df_chunk)
+    df_chunk = parse_data(df_chunk, sinan.cid10, sinan.year)
+    df_chunk = df_chunk.replace({pd.NA: None})
 
-        logger.info("Converting Parquet file into chunks")
+    # df_chunk = df_chunk.dropna(
+    #     subset=SINANUpload.REQUIRED_COLS, how="any"
+    # ) # it usually drops the entire chunk
```

Comment:
I don't know if its supposed to happen this way, but this dropna usually drops the entire chunk, few rows don't have None in some of the required columns

Reply:
Is this DROPNA necessary for the conversion to Parquet? If not, we should retain all rows independently of the fact that they may contain NAs. This `chunk_parquet_file` should not make decisions about data quality. It should already receive sanitized data.

Reply:
I thought that the parquet file would contain the raw data (the DBF converted to parquet, for instance) and the inserted rows of that file would be pointed in a separated table..

Reply:
You are right. The sanitization should be done as a separate step.